libtcmu_log.c

```diff
@@ -209,6 +209,34 @@ static void log_output(int pri, const char *msg, bool bypass)
 	}
 }
 
+struct log_buffer {
+	int pri;
+	char msg[LOG_MSG_LEN];
+};
+
+static darray(struct log_buffer) log_tmpbuf = darray_new();
+static pthread_mutex_t log_tmpbuf_lock = PTHREAD_MUTEX_INITIALIZER;
+
+static void buffer_log_messages(int pri, const char *fmt, va_list args)
+{
+	struct log_buffer lb;
+	char buf[LOG_MSG_LEN];
+	ssize_t len;
+
+	len = vsnprintf(buf, LOG_MSG_LEN, fmt, args);
+
+	/* Handle early log calls by config and deamon setup */
+	write(2, buf, len);
```

Comment:
Could you use the stdin/out/err def for this.

Reply:
Will update this.